### PR TITLE
Add CMakeLists.txt for six libraries

### PR DESCRIPTION
**Pull request by DemiMarie**
c-ares, dav1d, libraw, libvmaf, miniz, and tesseract-ocr are included.  dav1d is not very useful as the code generator crashes, so I had to disable wrapping `dav1d_open`.

### CODE REVIEW

File: contrib/dav1d/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "${dav1d_INCLUDEDIR}/dav1d/dav1d.h"
+    "${dav1d_INCLUDEDIR}/dav1d/data.h"
+  LIBRARY dav1d
+  FUNCTIONS
+    dav1d_data_create
+    dav1d_data_wrap
+    dav1d_data_wrap_user_data
+    dav1d_data_unref
+    dav1d_version
+    # dav1d_open
```
Comment (okunz) [inline contrib/dav1d/CMakeLists.txt:44]: possible left over or intentionally commented out?

Comment (DemiMarie) [reply contrib/dav1d/CMakeLists.txt:44]: Intentionally commented out because it causes the code generator to misbehave.

Comment (okunz) [reply contrib/dav1d/CMakeLists.txt:44]: That's a known issue with the our build environment at the moment. I suggest to remove the comment and include it directly. This is something we need to fix on our end.

**Review by cblichmann — CHANGES_REQUESTED** [review]
Can we split the libraries into different PRs?

> Reviewed 12 of 12 files at r1, 11 of 11 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/sandboxed-api/127#-:-MznczEs8pN98gukYI-D:b-tsv1r8)* status: all files reviewed, 4 unresolved discussions (waiting on @DemiMarie)

___
*[contrib/libraw/CMakeLists.txt, line 31 at r1](https://reviewable.io/reviews/google/sandboxed-api/127#-MzncdNa6S76u25dqUfp:-MzncdNa6S76u25dqUfq:bq2u6mv) ([raw file](https://github.com/google/sandboxed-api/blob/290f7a0fc2f094a3633214950c8f49f0c386cabf/contrib/libraw/CMakeLists.txt#L31)):*
> ```CMake
> pkg_check_modules(LIBRAW REQUIRED IMPORTED_TARGET libraw)
> 
> add_sapi_library(libraw_sapi
> ```

Isn't libraw also covered by #161 ?



_Code quote:_
```CMake
add_sapi_library(libraw_sapi
```
___
*[contrib/miniz/utils/utils_miniz.h, line 26 at r2](https://reviewable.io/reviews/google/sandboxed-api/127#-Mznb2jEBtMr20r7rfr8:-Mznb2jEBtMr20r7rfr9:bul1slz) ([raw file](https://github.com/google/sandboxed-api/blob/b5cd43e7ac16ecee79cadbe5b4941a75c627bd8a/contrib/miniz/utils/utils_miniz.h#L26)):*
> ```objc
> 
> namespace sapi {
> namespace util {
> ```

This can use nested-name specifiers: `namespace sapi::util {`



_Code quote:_
```objc
namespace sapi {
namespace util {
```
___
*[contrib/miniz/utils/utils_miniz.cc, line 69 at r2](https://reviewable.io/reviews/google/sandboxed-api/127#-MznbKyO8m3pNS8gcjxb:-MznbKyO8m3pNS8gcjxc:bvxpcx1) ([raw file](https://github.com/google/sandboxed-api/blob/b5cd43e7ac16ecee79cadbe5b4941a75c627bd8a/contrib/miniz/utils/utils_miniz.cc#L69)):*
> ```cpp
>   sapi::v::IntBase<size_t> outsize(0);
> 
>   // FIXME: this is vulnerable to a trivial DoS (memory exhaustion) by
> ```

We should set an rlimit in the sandbox executor then. This can be done in the Sandbox class by overriding `sapi::Sandbox::ModifyExecutor(sandbox2::Executor*)`.
___
*[sandboxed_api/tools/clang_generator/CMakeLists.txt, line 51 at r2](https://reviewable.io/reviews/google/sandboxed-api/127#-Mznc-Jm2w4KCe-ogGVz:-Mznc-Jm2w4KCe-ogGW-:bvrdul3) ([raw file](https://github.com/google/sandboxed-api/blob/b5cd43e7ac16ecee79cadbe5b4941a75c627bd8a/sandboxed_api/tools/clang_generator/CMakeLists.txt#L51)):*
> ```CMake
>   absl::strings
>   clang
>   clang-cpp
> ```

I think we need to find a different way on Fedora. LLVM suggests that using the "clang"-prefixed libraries is preferrable, so `clangFormat`, `clangTooling`, etc. should be used.

Perhaps we're just missing a `target_link_directories(sapi_generator ${LLVM_LIB_DIR})` ?



_Code quote:_
```CMake
  clang
  clang-cpp
```

**Pull request closed without merging**